File: Back/Pragmap/Pragmap.Domain/Entities/Customer.cs
using Pragmap.Domain.Common.Interfaces;
using System.ComponentModel.DataAnnotations;

namespace Pragmap.Domain.Entities
{
    public class Customer : IDatedEntity
    {
        public Guid Id { get; set; }

        [MaxLength(255)]
        public string Name { get; set; }

        public string Logo { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }

        public virtual ICollection<CustomerUser>? CustomerUsers { get; set; }

    }
}

File: Front/Pragmap/src/types/customer.ts
import type { ICustomerUser } from './customerUser'

export interface ICustomer {
	id: string
	name: string
	logo: string | null
	userIds: Array<string> | []
	createdAt: string
	updatedAt: string
	customerUsers: Array<ICustomerUser> | []
}
export interface IGetCustomer {
	id: string
	name: string
	logo: string | null
	createdAt: string
	customerUsers: Array<ICustomerUser> | []
}
export interface IPostCustomer {
	name: string
	logo: string | null
	userIds: Array<string> | []
}
export interface IPutCustomer {
	id: string
	name: string
	logo: string | null
	userIds: Array<string> | []
}
